test/checks/navigation/region.js

```diff
@@ -74,9 +66,75 @@ describe('region', function () {
+
+		assert.isFalse(checks.region.evaluate.apply(checkContext, checkArgs));
 		assert.equal(checkContext._relatedNodes.length, 1);
 	});
+
+	it('should return true if the non-region element is a script', function () {
+		var checkArgs = checkSetup('<div id="target"><script>axe.run()</script><div role="main">Content</div></div>');
+
+		assert.isTrue(checks.region.evaluate.apply(checkContext, checkArgs));
+	});
+
+	it('should considered aria labelled elements as content', function () {
+		var checkArgs = checkSetup('<div id="target"><div aria-label="axe-core logo"></div><div role="main">Content</div></div>');
+
+		assert.isFalse(checks.region.evaluate.apply(checkContext, checkArgs));
+		assert.deepEqual(checkContext._relatedNodes, [
+			fixture.querySelector('div[aria-label]')
+		]);
+	});
+
+	it('should allow native landmark elements', function () {
+		var checkArgs = checkSetup('<div id="target"><header>branding</header><main>Content </main><aside>stuff</aside><footer>copyright</footer></div>');
+
+		assert.isTrue(checks.region.evaluate.apply(checkContext, checkArgs));
+		assert.lengthOf(checkContext._relatedNodes, 0);
+	});
+
+	(shadowSupport.v1 ? it : xit)('should test Shadow tree content', function () {
+		var div = document.createElement('div');
+		var shadow = div.attachShadow({ mode: 'open' });
+		shadow.innerHTML = 'Some text';
+		var checkArgs = checkSetup(div);
+
+		assert.isFalse(checks.region.evaluate.apply(checkContext, checkArgs));
+		assert.deepEqual(checkContext._relatedNodes, [div]);
+	});
+
+	(shadowSupport.v1 ? it : xit)('should test slotted content', function () {
+		var div = document.createElement('div');
+		div.innerHTML = 'Some content';
+		var shadow = div.attachShadow({ mode: 'open' });
+		shadow.innerHTML = '<div role="main"><slot></slot></div>';
+		var checkArgs = checkSetup(div);
+
+		assert.isTrue(checks.region.evaluate.apply(checkContext, checkArgs));
+		assert.lengthOf(checkContext._relatedNodes, 0);
+	});
+
+	(shadowSupport.v1 ? it : xit)('should ignore skiplink targets inside shadow trees', function () {
+		var div = document.createElement('div');
+		div.innerHTML = '<a href="#foo">skiplink</a><div>Content</div>';
+
+		var shadow = div.querySelector('div').attachShadow({ mode: 'open' });
+		shadow.innerHTML = '<div role="main" id=#foo"><slot></slot></div>';
+		var checkArgs = checkSetup(div);
+
+		assert.isFalse(checks.region.evaluate.apply(checkContext, checkArgs));
+		assert.deepEqual(checkContext._relatedNodes, [div.querySelector('a')]);
+	});
+
+	(shadowSupport.v1 ? it : xit)('should find the skiplink in shadow DOM', function () {
+		var div = document.createElement('div');
+		div.innerHTML = '<span id="foo">Content!</span>';
+		var shadow = div.attachShadow({ mode: 'open' });
+		shadow.innerHTML = '<a href="#foo">skiplink</a><div role="main"><slot></slot></div>';
```

Comment:
The href is inside the shadow DOM, but the IDref is in the light DOM...? How does that work?